File: Web/src/app/models/topic.ts
import { Attendance } from "./attendance";
import { Feedback } from "./feedback";
import { Rating } from "./rating";
import { TopicComment } from "./topicComment";

export class Topic {
    public id: string;
    public name: string;
    public demands: string[];
    public description: string;
    public owner: string;
    public attendees: Attendance[];
    public comments: TopicComment[];
    public feedback: Feedback[];
    public ratings: Rating[];
    public roomId: string;
    public slotId: string;

    constructor(owner: string = null) {
        this.owner = owner;
    }
}

File: Common/Entities/Topic.cs
using System;
using System.Collections.Generic;

namespace openspace.Common.Entities
{
    public class Topic
    {
        public ICollection<Attendance> Attendees { get; set; } = new List<Attendance>();

        public ICollection<string> Demands { get; set; } = new List<string>();

        public string Description { get; set; }

        public ICollection<Feedback> Feedback { get; set; } = new List<Feedback>();

        public ICollection<TopicComment> Comments { get; set; } = new List<TopicComment>();

        public string Id { get; set; } = Guid.NewGuid().ToString();

        public string Name { get; set; }

        public string Owner { get; set; }

        public ICollection<Rating> Ratings { get; set; } = new List<Rating>();

        public string RoomId { get; set; }

        public string SlotId { get; set; }
    }
}
